### DISCUSSION TITLE
CoreWCF v1.0.0

### DISCUSSION THREAD

**Discussion by mconnew** [CoreWCF/CoreWCF#631] [Releases]
# Release Notes

## Package Info

[CoreWCF.Primitives 1.0.0](https://www.nuget.org/packages/CoreWCF.Primitives/1.0.0)  
[CoreWCF.Http 1.0.0](https://www.nuget.org/packages/CoreWCF.Http/1.0.0)  
[CoreWCF.NetTcp 1.0.0](https://www.nuget.org/packages/CoreWCF.NetTcp/1.0.0)  
[CoreWCF.WebHttp 1.0.0](https://www.nuget.org/packages/CoreWCF.WebHttp/1.0.0)  
[CoreWCF.ConfigurationManager 1.0.0](https://www.nuget.org/packages/CoreWCF.ConfigurationManager/1.0.0)

## .NET Compatibility
This release depends on .NET Standard 2.0 and runs on any .NET version which supports .NET Standard 2.0. This release supports .NET Framework 4.6.2 and above, .NET Core 3.1, .NET 5, and .NET 6. It is built on top of ASP.NET Core and has been tested and runs on all currently supported versions of ASP.NET Core up to 6.0.

## New Features
* Custom Binding support for Configuration (@kbrowdev)

## What's Changed
* Fixed some OpenAPI documentation generation issues by @JonathanHopeDMRC in https://github.com/CoreWCF/CoreWCF/pull/617
* Fixed Url base address for NetTcp when listening on IPv6 by @FreeAndNil in https://github.com/CoreWCF/CoreWCF/pull/609
* Fixed Get only collection support in WSDL generation by @roend83 in https://github.com/CoreWCF/CoreWCF/pull/616
* Added support for MessagePropertyAttribute and fixed support for MessageHeaderArrayAttribute by @mconnew in https://github.com/CoreWCF/CoreWCF/pull/623
* Enable compression over NetTcp by @jonlouie in https://github.com/CoreWCF/CoreWCF/pull/627
* Fixing bad requests to return a 400 response to match WCF by @mconnew in https://github.com/CoreWCF/CoreWCF/pull/626
* Fixed deriving SecurityTokenManager as abstract method was internal by @gougeman in https://github.com/CoreWCF/CoreWCF/pull/630
* Added configuration support for custom bindings by @kbrowdev in https://github.com/CoreWCF/CoreWCF/pull/578

## New Contributors
* @jonlouie made their first contribution in https://github.com/CoreWCF/CoreWCF/pull/627
* @gougeman made their first contribution in https://github.com/CoreWCF/CoreWCF/pull/630
* @kbrowdev made their first contribution in https://github.com/CoreWCF/CoreWCF/pull/578

**Full Changelog**: https://github.com/CoreWCF/CoreWCF/compare/v1.0.0-preview2...v1.0.0

## Feedback
Your feedback is important and appreciated. We've created a discussion item at https://github.com/CoreWCF/CoreWCF/discussions/631 for your questions and comments.

<hr /><em>This discussion was created from the release <a href='https://github.com/CoreWCF/CoreWCF/releases/tag/v1.0.0'>CoreWCF v1.0.0</a>.</em>

**Comment by absinghal** [comment]
Congratulations!

**Comment by mconnew** [comment]
# Release Notes for v1.0.1

## Package Info

[CoreWCF.Primitives 1.0.1](https://www.nuget.org/packages/CoreWCF.Primitives/1.0.1)  
[CoreWCF.Http 1.0.1](https://www.nuget.org/packages/CoreWCF.Http/1.0.1)  
[CoreWCF.NetTcp 1.0.1](https://www.nuget.org/packages/CoreWCF.NetTcp/1.0.1)  
[CoreWCF.WebHttp 1.0.1](https://www.nuget.org/packages/CoreWCF.WebHttp/1.0.1)  
[CoreWCF.ConfigurationManager 1.0.1](https://www.nuget.org/packages/CoreWCF.ConfigurationManager/1.0.1)

## .NET Compatibility
This release depends on .NET Standard 2.0 and runs on any .NET version which supports .NET Standard 2.0. This release supports .NET Framework 4.6.2 and above, .NET Core 3.1, .NET 5, and .NET 6. It is built on top of ASP.NET Core and has been tested and runs on all currently supported versions of ASP.NET Core up to 6.0.

## New Features
None

## What's Changed
* Prevent CoreWCF source generator from interfering with ASP.NET Core MVC Controller when using FromService attribute by @g7ed6e in #636
* Reduced the number of scenarios where AllowSynchronousIO is needed by @mconnew in #648

**Full Changelog**: https://github.com/CoreWCF/CoreWCF/compare/v1.0.0...v1.0.1

## Feedback
Your feedback is important and appreciated. Please use the discussion item for the v1.0 release at https://github.com/CoreWCF/CoreWCF/discussions/631 for your questions and comments.
